lib/double_linked_list.rb

```diff
@@ -0,0 +1,119 @@
+class DoubleNode
+    attr_accessor :data, :next, :previous
+  
+    def initialize(data, next_node, previous_node)
+        @data = data
+        @next = next_node
+        @previous = previous_node
+    end
+  end
+
+# Defines the doubly linked list
+class DoublyLinkedList
+    attr_reader :tail, :head
+    def initialize
+        @head = nil
+        @tail = nil
+    end
+
+    def add_first(data)
+        if @head.nil?
+            @head = DoubleNode.new(data, @head, nil)
+            @tail = @head
+        elsif @head.next.nil?
+            new_head = DoubleNode.new(data, @head, nil)
+            @tail = @head
```

Comment:
This line shouldn't be needed if `@head.next.nil?` then tail should point to the same node as head.